packages/dataviews/src/dataviews-layouts/utils/item-click-wrapper.tsx

```diff
@@ -69,12 +74,25 @@ export function ItemClickWrapper< Item >( {
 
 	// If we have a renderItemLink, use it
 	if ( renderItemLink ) {
-		return renderItemLink( {
+		const renderedElement = renderItemLink( {
 			item,
 			className: `${ className } ${ className }--clickable`,
 			...extraProps,
 			children,
 		} );
+
+		// Clone the element and enhance onClick to stop propagation
+		return cloneElement( renderedElement, {
+			onClick: ( event: React.MouseEvent ) => {
```

Comment:
It'd be good to also cover keyboard clicking, like [we do](https://github.com/WordPress/gutenberg/blob/56c6552dd682a46724d7cdf9dfc5ebe47098d22b/packages/dataviews/src/dataviews-layouts/utils/item-click-wrapper.tsx#L32-L42) when we render our own handler.